src/allotropy/parsers/utils/values.py

```diff
@@ -42,3 +52,66 @@ def assert_not_none(
 
 def value_or_none(value: str) -> Optional[str]:
     return value.strip() or None
+
+
+def df_to_series(df: pd.DataFrame, msg: str) -> pd.Series:
+    n_rows, _ = df.shape
+    if n_rows == 1:
+        return pd.Series(df.iloc[0], index=df.columns)
+    raise AllotropeConversionError(msg)
+
+
+def assert_not_empty_df(df: pd.DataFrame, msg: str) -> pd.DataFrame:
+    if df.empty:
+        raise AllotropeConversionError(msg)
+    return df
+
+
+def str_from_series(
+    data: pd.Series, key: str, default: Optional[str] = None
```

Comment:
The `default` parameter is unused in callers.

And if we did want a default, it would be better to have a different signature (similar to `assert_str_from_series`) in which default wasn't `Optional` and thus neither was the return type.